client/src/components/googleMap/googleMap.jsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Loader} from '@googlemaps/js-api-loader';
+import './googleMap.sass';
+
+/* eslint-disable import/no-unresolved */
+const config = require('../../../../server/config');
+/* eslint-enable import/no-unresolved */
+
+const loader = new Loader({
+	apiKey: config.mapsAPIKey,
+	version: "weekly"
+});
+
+export class GoogleMap extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			markers: [],
+			location: null
+		};
+
+		this.mapRef = React.createRef();
+		this.createDisplayMap = this.createDisplayMap.bind(this);
+		this.createSightingMap = this.createSightingMap.bind(this);
+		this.createMarkers = this.createMarkers.bind(this);
+		this.addMarker = this.addMarker.bind(this);
+		this.showMarkers = this.showMarkers.bind(this);
+		this.hideMarkers = this.hideMarkers.bind(this);
+		this.deleteMarkers = this.deleteMarkers.bind(this);
+		this.updateLocationMarker = this.updateLocationMarker.bind(this);
+		this.addEventListener = this.addEventListener.bind(this);
+	};
+
+	//Creates a new marker on the map, updates state with the new marker
+	addMarker(location, map) {
+		const newMarker = new google.maps.Marker({
+			position: location,
+		});
+
+		let marker = this.state.markers;
+		marker.push(newMarker);
+		this.setState({ markers: marker });
+		this.showMarkers(map);
+	}
+
+	//Shows all markers in state on the map
+	showMarkers(map) {
+		console.log("running show markers");
+		const markers = this.state.markers;
+		for (let i = 0; i < markers.length; i++) {
+			markers[i].setMap(map);
+		}
+	}
+
+	//Removes all markers from the map, keeps in state
+	hideMarkers() {
+		console.log("running hide markers");
+		this.showMarkers(null);
+	}
+
+	//Removes all markers from map and clears state
+	deleteMarkers() {
+		console.log("running delete markers");
+		this.hideMarkers();
+		this.setState({ markers: [] });
+	}
+
+	//Updates the marker location on the map and removes the previous markers
+	updateLocationMarker(location, map) {
+		console.log("running update location markers");
+		this.deleteMarkers();
+		this.addMarker(location, map);
+	}
+
+	createMarkers(map) {
+		console.log("running createMarkers");
+		const sightings = this.props.sightings;
+		if (!sightings || sightings.length < 0) return null;
+
+		sightings.forEach(element => {
+			const defaultLocation = { lat: 45.63177710291909, lng: -79.71027154237892 };	
+			
+			let location;
+			try {
+				location = JSON.parse(element.location);
+			}
+			catch(err) {
+				location = defaultLocation;
+			}
+
+			const marker = new google.maps.Marker({
+				position: location,
+				map: map
+			});
+
+			//Some test DB entries don't have users assigned to them
+			let popUpInfo;
+			if (element.spottedBy) {
+				popUpInfo = '<div><h1>' + element.animal + '</h1><h3>' + element.spottedBy + '</h3></div>';
```

Comment:
Make these with template strings. Can also combine line 100 and 102 into a single thing with a ternary